rasa/nlu/extractors/entity_synonyms.py

```diff
@@ -1,41 +1,65 @@
 import os
-from typing import Any, Dict, List, Optional, Text, Type
+from typing import Any, Dict, List, Optional, Text
 
-from rasa.nlu.components import Component
+from rasa.engine.graph import GraphComponent, ExecutionContext
 from rasa.shared.constants import DOCS_URL_TRAINING_DATA
 from rasa.shared.nlu.constants import ENTITIES, TEXT
-from rasa.nlu.config import RasaNLUModelConfig
-from rasa.nlu.extractors.extractor import EntityExtractor
-from rasa.nlu.model import Metadata
 from rasa.shared.nlu.training_data.training_data import TrainingData
 from rasa.shared.nlu.training_data.message import Message
 from rasa.nlu.utils import write_json_to_file
+from rasa.nlu.extractors.extractor import EntityExtractorMixin
 import rasa.utils.io
 import rasa.shared.utils.io
+from rasa.nlu.extractors._entity_synonyms import EntitySynonymMapper
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
 
+# This is a workaround around until we have all components migrated to `GraphComponent`.
+EntitySynonymMapper = EntitySynonymMapper
 
-class EntitySynonymMapper(EntityExtractor):
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        return [EntityExtractor]
+
+class EntitySynonymMapperComponent(GraphComponent, EntityExtractorMixin):
+    """Maps entities to their synonyms if they appear in the training data."""
+
+    SYNONYM_FILENAME = "synonyms.json"
 
     def __init__(
         self,
-        component_config: Optional[Dict[Text, Any]] = None,
+        config: Optional[Dict[Text, Any]] = None,
+        model_storage: ModelStorage = None,
+        resource: Resource = None,
```

Comment:
no need for defaults - these will always be passed
```suggestion
        config: Dict[Text, Any],
        model_storage: ModelStorage,
        resource: Resource,
```